airflow/io/__init__.py

```diff
@@ -86,7 +93,13 @@ def get_fs(
         raise ValueError(f"No filesystem registered for scheme {scheme}") from None
 
     options = storage_options or {}
-    return fs(conn_id, options)
+    # MyPy does not recognize dynamic parameters inspection when we call the method, and we have to do
+    # it for compatibility reasons with already released providers, that's why we need to ignore
+    # mypy errors here
+    parameters = inspect.signature(fs).parameters
+    if len(parameters) == 1:
+        return fs(conn_id)  # type: ignore[call-arg]
+    return fs(conn_id, options)  # type: ignore[call-arg]
```

Comment:
Should we add a test for the old signature to verify

Reply:
Let me see. It's new code for me :)

Reply:
Isn't the typical pythonic way to do this with a try catch? Inspect is expensive.

Reply:
yeah - it's a bit more expensive, but It happens only once at discovery. 

I thought about it too but I decided dointg it more explicitly is better. It is much better to handle (potential) patterns where we add more parameters (which might happen). In this case we will just add `if len() ==2`  - it would be difficult to distinguish those cases with exception.

Reply:
Also, this is pretty common pattern in Airflow already. We use inspect and looking at parameters and comparing signatures of methods in quite a numer of places already. I personally see it as much more explicit. But of course nothing prevents us from changing it in the future if we agree it's a better solution, this part is internal and can be changed any time.